x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -1100,6 +1176,224 @@ export class RulesClient {
     );
   }
 
+  public async bulkEdit<Params extends RuleTypeParams>({
+    filter,
+    actions,
+    paramsModifier,
+  }: BulkEditOptions<Params>): Promise<{
+    rules: Array<SanitizedRule<Params>>;
+    errors: BulkEditError[];
+  }> {
+    let authorizationTuple;
+    try {
+      authorizationTuple = await this.authorization.getFindAuthorizationFilter(
+        AlertingAuthorizationEntity.Rule,
+        alertingAuthorizationFilterOpts
+      );
+    } catch (error) {
+      this.auditLogger?.log(
+        ruleAuditEvent({
+          action: RuleAuditAction.BULK_EDIT,
+          error,
+        })
+      );
+      throw error;
+    }
+    const { filter: authorizationFilter } = authorizationTuple;
+    const qNodeFilter =
+      (authorizationFilter && filter
+        ? nodeBuilder.and([
+            typeof filter === 'string' ? esKuery.fromKueryExpression(filter) : filter,
+            authorizationFilter as KueryNode,
+          ])
+        : authorizationFilter) ?? filter;
+
+    const { aggregations, total } = await this.unsecuredSavedObjectsClient.find<
+      RawRule,
+      RuleBulkEditAggregation
+    >({
+      filter: qNodeFilter,
+      page: 1,
+      perPage: 0,
+      type: 'alert',
+      aggs: {
+        alertTypeId: {
+          multi_terms: {
+            terms: [{ field: 'alert.alertTypeId' }, { field: 'alert.consumer' }],
+          },
+        },
+      },
+    });
+
+    if (total > MAX_RULES_NUMBER_FOR_BULK_EDIT) {
+      throw Error(`More than ${MAX_RULES_NUMBER_FOR_BULK_EDIT} rules matched for bulk edit`);
+    }
+    const buckets = aggregations?.alertTypeId.buckets;
+
+    if (buckets === undefined) {
+      throw Error('Aggregation by alertTypeId|consumer is empty');
+    }
+
+    await pMap(buckets, async ({ key: [ruleType, consumer] }) => {
```

Comment:
initially, I didn't put it there, as it seemed there are shouldn't be a lot of concurrent actions based on aggregation results. But it might be good thing to put it there.